Documentation/Development.md

```diff
@@ -6,27 +6,232 @@
+:: configure path - we need the Git tools in the path
+path %ProgramFiles%\Git\usr\bin;%PATH%
+:: ensure that python dependencies are available
+python -m pip install --user numpy six
+:: configure X10/TensorFlow
+set TF_ENABLE_XLA=1
+set TF_NEED_ROCM=0
+set TF_NEED_CUDA=0
+set TF_CUDA_COMPUTE_CAPABILITIES=7.5
+set CC_OPT_FLAGS="/arch:AVX /D_USE_MATH_DEFINES"
+set TF_OVERRIDE_EIGEN_STRONG_INLINE=1
+.\tensorflow\configure.py
+:: build
+set BAZEL_SH=%ProgramFiles%\Git\usr\bin\bash.exe
+set BAZEL_VC=%VCINSTALLDIR%
+bazel --output_user_root %CD%/caches/bazel/tensorflow build -c opt --copt /D_USE_MATH_DEFINES --define framework_shared_object=false --config short_logs --nocheck_visibility //tensorflow:tensorflow //tensorflow:tensorflow_dll_import_lib //tensorflow/compiler/tf2xla/xla_tensor:x10 //tensorflow/compiler/tf2xla/xla_tensor:x10_dll_import_lib
+:: terminate bazel daemon
+bazel --output_user_root %CD%/caches/bazel/tensorflow shutdown
+
+:: package
+set DESTDIR=%CD%\Library\tensorflow-windows-%VSCMD_ARG_TGT_ARCH%\tensorflow-2.4.0
+
+md %DESTDIR\usr\bin
+copy tensorflow\bazel-bin\tensorflow\tensorflow.dll %DESTDIR%\usr\bin\
+copy tensorflow\bazel-bin\tensorflow\compiler\tf2xla\xla_tensor\x10.dll %DESTDIR%\usr\bin\
+
+md %DESTDIR%\usr\lib
+copy tensorflow\bazel-out\%VSCMD_ARG_TGT_ARCH%_windows-opt\bin\tensorflow\tensorflow.lib %DESTDIR%\usr\lib\
+copy tensorflow\bazel-out\%VSCMD_ARG_TGT_ARCH%_windows-opt\bin\tensorflow\compiler\tf2xla\xla_tensor\x10.lib %DESTDIR%\usr\lib\
+
+md %DESTDIR%\usr\include\tensorflow\c
+copy tensorflow\tensorflow\c\c_api.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\c_api_experimental.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\tf_attrtype.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\tf_datatype.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\tf_file_statistics.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\tf_status.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\tf_tensor.h %DESTDIR%\usr\include\tensorflow\c\
+
+md %DESTDIR%\usr\include\tensorflow\c\eager
+cp tensorflow\tensorflow\c\eager\c_api.h %DESTDIR%\usr\include\tensorflow\c\eager\
+
+md %DESTDIR%\usr\include\x10
+copy swift-apis\Sources\x10\swift_bindings\device_wrapper.h %DESTDIR%\usr\include\x10\
+copy swift-apis\Sources\x10\swift_bindings\xla_tensor_tf_ops.h %DESTDIR%\usr\include\x10\
+copy swift-apis\Sources\x10\swift_bindings\xla_tensor_wrapper.h %DESTDIR%\usr\include\x10\
+
+md %DESTDIR%\usr\share
+copy tensorflow\bazel-out\%VSCMD_ARG_TGT_ARCH%_windows-opt\bin\tensorflow\tensorflow_filtered_def_file.def %DESTDIR%\usr\share
+```
+</details>
+
+<details>
+    <summary>macOS/Linux Build Script</summary>
+
+```bash
+# clone swift-apis
+git clone git://github.com/tensorflow/swift-apis
+# checkout tensorflow
+git clone --depth 1 --no-tags git://github.com/tensorflow/tensorflow
+git -C tensorflow checkout refs/tags/v2.4.0
```

Comment:
These two steps don't seem to work, because tags aren't cloned:
```console
$ git clone --depth 1 --no-tags git://github.com/tensorflow/tensorflow
$ git -C tensorflow checkout refs/tags/v2.4.0
Cloning into 'tensorflow'...
remote: Enumerating objects: 23521, done.
remote: Counting objects: 100% (23521/23521), done.
remote: Compressing objects: 100% (17257/17257), done.
remote: Total 23521 (delta 8698), reused 10244 (delta 5705), pack-reused 0
Receiving objects: 100% (23521/23521), 56.29 MiB | 722.00 KiB/s, done.
Resolving deltas: 100% (8698/8698), done.
Updating files: 100% (24068/24068), done.
error: pathspec 'refs/tags/v2.4.0' did not match any file(s) known to git
```

Reply:
We could also just change to `git -C tensorflow checkout refs/heads/r2.4` which matches what CMake currently does.